config/gcp.spc

```diff
@@ -20,6 +20,9 @@ connection "gcp" {
   # If not set, no impersonation is done.
   #impersonate_service_account = "YOUR_SERVICE_ACCOUNT"
 
+  # `ignore_error_messages` (optional) - List of additional GCP error message pattern to ignore for all queries.
+  #  ignore_error_messages = ["API has not been used", "other regex"]
```

Comment:
@ParthaI Is there a more exact regex (using `^`, `$`) we should add as an example here, in particular for the "API has not been used" case? We can probably remove `other regex` as it's not helpful.

Reply:
This is a really good addition to handle cases where, google uses the same 403 error code when - API are not enabled and when [Quota exceeded](https://cloud.google.com/bigquery/docs/troubleshoot-quotas#overview). An exact regex mapping example to an error message would sure be helpful.

Reply:
For reference, this is reported as this:

```
ERROR: googleapi: Error 403: Cloud DNS API has not been used in project XXX before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/dns.go…
Details:
[
  {
    "@type": "type.googleapis.com/google.rpc.Help",
    "links": [
      {
        "description": "Google developers console API activation",
        "url": "https://console.developers.google.com/apis/api/dns.googleapis.com/overview?project=XXX"
      }
    ]
  },
  {
    "@type": "type.googleapis.com/google.rpc.ErrorInfo",
    "domain": "googleapis.com",
    "metadata": {
      "consumer": "projects/XXX",
      "service": "dns.googleapis.com"
    },
    "reason": "SERVICE_DISABLED"
  }
]
, accessNotConfigured (SQLSTATE HV000)
```

So I think the current example with `API has not been used` would be enough.

Reply:
@cbruno10, I have added the example for the exact regex (using ^, $) i.e `ignore_error_messages = ["^.*API\\s*has\\s*not\\s*been\\s*used.*$"]`. Also we can pass a message just like a string format, in both of the case the code changes are working fine.

NOTE: In the .spc file, we should use `\\` instead of `\` for the regex pattern to avoid errors related to escape characters. Otherwise, parsing the .spc file might result in an error like "Invalid escape sequence: The symbol "d" is not a valid escape sequence selector."